loader/icd_dispatch.c

```diff
@@ -99,6 +99,17 @@ clGetDeviceIDs(cl_platform_id   platform,
 {
     // initialize the platforms (in case they have not been already)
     khrIcdInitialize();
+
+    // determine the platform to use from the device_type specified
```

Comment:
This is technically OK since the spec says that behavior is implementation-defined when `platform` is `NULL`, but I do wonder if this is right direction.  In general, I'd like to "thin out" these aspects of the ICD loader, to the point where it could be automatically generated in many cases (see: #73, #76).  Also, not that if we added this now, there would be some ICD loaders that return NULL and an error with a NULL platform, and others where things magically work, which seems undesirable.

Unless there is a strong reason to add this behavior, I'd prefer to revert this change.

Reply:
Reverted the part of the change which was overriding platform with platform[0].

Reply:
I still see a change here that I'd prefer to revert.